File: Models/campbell_SoilTemp/Campbell/src/java/Campbell/campbellAuxiliary.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class campbellAuxiliary
{
    private Double [] minSoilTemp;
    
    public campbellAuxiliary() { }
    
    public campbellAuxiliary(campbellAuxiliary toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            minSoilTemp = new Double[toCopy.getminSoilTemp().length];
        for (int i = 0; i < toCopy.getminSoilTemp().length; i++)
        {
            minSoilTemp[i] = toCopy.getminSoilTemp()[i];
        }
        }
    }
    public Double [] getminSoilTemp()
    { return minSoilTemp; }

    public void setminSoilTemp(Double [] _minSoilTemp)
    { this.minSoilTemp= _minSoilTemp; } 
    
}

File: Models/campbell_SoilTemp/Campbell/src/cs/Campbell/campbellAuxiliary.cs
using System;
using System.Collections.Generic;

public class CampbellAuxiliary 
{
    private double[] _minSoilTemp;
    
    /// <summary>
    /// Constructor of the campbellAuxiliary component")
    /// </summary>  
    public campbellAuxiliary() { }
    
    
    public campbellAuxiliary(campbellAuxiliary toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            minSoilTemp = new double[toCopy.minSoilTemp.Length];
            for (int i = 0; i < toCopy.minSoilTemp.Length; i++)
                { minSoilTemp[i] = toCopy.minSoilTemp[i]; }
    
        }
    }
    public double[] minSoilTemp
    {
        get { return this._minSoilTemp; }
        set { this._minSoilTemp= value; } 
    }
}